src/main/scala/org/constellation/snapshotstreaming/opensearch/schema/AddressBalance.scala

```diff
@@ -0,0 +1,27 @@
+package org.constellation.snapshotstreaming.opensearch.schema
+
+import java.util.Date
+
+import io.circe.Encoder
+import io.circe.generic.semiauto._
+
+final case class AddressBalance(
+  address: String,
+  balance: Long,
+  snapshotHash: String,
+  snapshotOrdinal: Long,
+  timestamp: Date
+)
+
+object AddressBalance {
+
+  implicit val dateEncoder: Encoder[Date] =
+    Encoder.encodeString.contramap(date => date.toInstant.toString)
+
+  implicit val snapshotEncoder: Encoder[AddressBalance] = deriveEncoder
+
+  implicit class AddressBalanceOps(balance: AddressBalance) {
+    def docId = s"${balance.address}${balance.snapshotOrdinal}"
+  }
```

Comment:
you can simplify this as a function in the case class:
```
case class AddressBalance(
   address: String,
   balance: Long,
   snapshotHash: String,
   snapshotOrdinal: Long,
   timestamp: Date
 ) {
  def docId = s"${balance.address}${balance.snapshotOrdinal}"
 }
```
you don't need an implicit class for this